core/src/ids.rs

```diff
@@ -98,17 +98,26 @@ macro_rules! declare_id {
 
         impl core::fmt::Display for $name {
             fn fmt(&self, f: &mut core::fmt::Formatter) -> core::fmt::Result {
-                let mut end = self.0.len();
+                let len = self.0.len();
+                let mut e1 = len / 2;
+                let mut s2 = len / 2;
 
                 if let Some(precision) = f.precision() {
-                    if precision > end {
-                        return Err(core::fmt::Error);
-                    }
+                    let precision = precision.min(len / 2);
 
-                    end = precision;
+                    e1 = precision;
+                    s2 = len - precision;
                 };
 
-                write!(f, "0x{}", hex::encode(&self.0[..end]))
+                let p1 = hex::encode(&self.0[..e1]);
+                let p2 = hex::encode(&self.0[s2..]);
+                let sep = e1.ne(&s2).then_some("..").unwrap_or_default();
```

Comment:
I know you love `then_some` 😄 
But `e1.ne(&s2)` doesn't work for odd `len`.

```rust
let mut sep = "";
if matches!(f.precision(), Some(precision) if precision < len/2) {
    // ...
    sep = "..";
}

Reply:
As mentioned above, it works due to intervals specification